encryption.go

```diff
@@ -0,0 +1,413 @@
+package winrm
+
+import (
+	"bytes"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"regexp"
+	"strconv"
+	"strings"
+
+	ntlmssp "github.com/CalypsoSys/bobntlmssp"
+	ntlmhttp "github.com/CalypsoSys/bobntlmssp/http"
+	"github.com/masterzen/winrm/soap"
+)
+
+type Encryption struct {
+	ntlm           *ClientNTLM
+	SIXTEN_KB      int
+	MIME_BOUNDARY  []byte
+	protocol       string
+	protocolString []byte
+	httpClient     *http.Client
+	ntlmClient     *ntlmssp.Client
+	ntlmhttp       *ntlmhttp.Client
+}
+
+const (
+	SixteenKB      = 16384
+	MimeBoundary   = "--Encrypted Boundary"
+	defaultCipher  = "RC4-HMAC-NTLM"
+	BoundaryLength = len(MimeBoundary)
+)
+
+/*
+Encrypted Message Types
+When using Encryption, there are three options available
+	1. Negotiate/SPNEGO
+	2. Kerberos
+	3. CredSSP
+
+	protocol: The protocol string used for the particular auth protocol
+
+	The auth protocol used, will determine the wrapping and unwrapping method plus
+	the protocol string to use. Currently only NTLM is supported
+
+	based on the python code from https://pypi.org/project/pywinrm/
+
+	uses the most excellent NTLM library from https://github.com/bodgit/ntlmssp
+	the modified vesion only differs in not outputing raw http request and response
+*/
+func NewEncryption(protocol string) (*Encryption, error) {
+	encryption := &Encryption{
+		ntlm:          &ClientNTLM{},
+		SIXTEN_KB:     SixteenKB,
+		MIME_BOUNDARY: []byte(MimeBoundary),

```

Comment:
I don't think we need to have them per encryption since those are constants.
We could just refer to the constants when needed (unless I missed something)

Reply:
removed for const values